### PR TITLE
Migrates user settings to preferences

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7024
Resolves https://github.com/brave/brave-browser/issues/7101
Resolves https://github.com/brave/brave-browser/issues/9547

DB schema change here https://github.com/brave/rewards/pull/8

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- start from master
- enable rewards
- change values for AC settings
- claim grant and do some tips
- close browser
- switch to this PR
- start browser again
- make sure that no data was lost (AC settings and summary report)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 1]
```diff
@@ -846,11 +847,23 @@ void RewardsDOMHandler::SaveSetting(const base::ListValue* args) {
...
+      }
+
+      rewards_service_->SetPublisherMinVisitTime(int_value);
     }
 
     if (key == "contributionMinVisits") {
-      rewards_service_->SetPublisherMinVisits(std::stoul(value));
+      int int_value;
+      if (!base::StringToInt(value, &int_value)) {
+        LOG(ERROR) << "Min time was not converted to int";
```
Comment (emerick) [inline browser/ui/webui/brave_rewards_page_ui.cc:862]: Should be "min visits"

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
+	/* Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.h:1]: Nit: Looks like this got indented by mistake.

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.h:1]: fixed

File: browser/ui/webui/brave_rewards_page_ui.cc (hunk 1 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_page_ui.cc:862]: fixed

**Comment by NejcZdovc** [comment]
CI failed, macos (interruption), restarting

**Pull request merged**